### PR TITLE
feat: index class issuers

### PR DESCRIPTION
**Pull request by wgwz**
Closes: https://github.com/regen-network/indexer/issues/30

### CODE REVIEW

**Comment by wgwz** [comment]
Screen Shot 2023-08-21 at 9 22 56 PM

This PR adds the ability for two queries on the indexer graphql api:

1. fetch all issuers by credit class id, i.e.:
```graphql
query IssuersByCreditClassId($creditClassId: String) {
  allCreditClasses(condition: {latest: true, creditClassId: $creditClassId}) {
    nodes {
      blockHeight
      creditClassId
      issuerAddress
    }
  }
}
```

2. fetch all credit classes by issuer address:
```graphql
query CreditClassesByIssuer($issuerAddress: String) {
  allCreditClasses(condition: {latest: true, issuerAddress: $issuerAddress}) {
    nodes {
      blockHeight
      creditClassId
      issuerAddress
    }
  }
}
```

File: index_classes.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+def fetch_issuers(height, class_id):
+    resp = requests.get(
+        f"{os.environ['REGEN_API']}/regen/ecocredit/v1/classes/{class_id}/issuers",
+        headers={"x-cosmos-block-height": str(height)},
+    )
+    resp.raise_for_status()
+    return resp.json()["issuers"]
+
+
+def _index_classes(pg_conn, _client, _chain_num):
```
Comment (ryanchristo) [inline index_classes.py:19]: This function indexes credit class issuers, not credit classes themselves. We should make that clear.

```suggestion
def _class_issuers(pg_conn, _client, _chain_num):
```

File: sql/V1_10__credit_classes.sql [hunk 2]
```diff
@@ -0,0 +1,28 @@
+CREATE TABLE IF NOT EXISTS
+  credit_classes (
```
Comment (ryanchristo) [inline sql/V1_10__credit_classes.sql:2]: This table is not storing credit classes, only class issuers. We should make this more clear.

```suggestion
  class_issuers (
```

File: index_classes.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+                        normalize["class_id"],
+                        issuer,
+                    )
+                    _cur.execute(
+                        insert_text,
+                        row,
+                    )
+                    logger.info(_cur.statusmessage)
+                pg_conn.commit()
+                logger.info("credit class issuers updated..")
```
Comment (ryanchristo) [inline index_classes.py:100]: Only one issuer is being added here.

```suggestion
                logger.info("credit class issuer added..")
```

File: sql/V1_10__credit_classes.sql [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    TYPE TEXT NOT NULL,
+    block_height BIGINT NOT NULL,
+    tx_idx SMALLINT NOT NULL,
+    msg_idx SMALLINT NOT NULL,
+    chain_num SMALLINT NOT NULL,
+    TIMESTAMP timestamptz,
+    tx_hash TEXT NOT NULL,
+    credit_class_id TEXT NOT NULL,
+    issuer_address TEXT NOT NULL,
```
Comment (ryanchristo) [inline sql/V1_10__credit_classes.sql:11]: I think we should be consistent with the on-chain field names, i.e. `class_id` and `issuer` here.

```suggestion
    class_id TEXT NOT NULL,
    issuer TEXT NOT NULL,
```

File: index_classes.py [hunk 5]
```diff
@@ -0,0 +1,108 @@
+import logging
+import os
+import textwrap
+import requests
+from utils import PollingProcess, events_to_process
+
+logger = logging.getLogger(__name__)
+
+
+def fetch_issuers(height, class_id):
```
Comment (ryanchristo) [inline index_classes.py:10]:
```suggestion
def fetch_class_issuers(height, class_id):
```

File: index_classes.py (hunk 3 above)
Comment (wgwz) [reply index_classes.py:100]: plural makes sense, we commit after all inserts are taken care of

**Comment by wgwz** [comment]
@ryanchristo done! for most part i made exactly the changes you suggested but some small differences. LMK how it looks now

**Review by ryanchristo — APPROVED** [review]
utACK. Updates look great. Nice work!

**Pull request merged**